spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicSchedulerWrapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.scheduling.TaskScheduler;
+
+/**
+ * A wrapper class for a {@link TaskScheduler} to use for scheduling container resumption
+ * when a partition has been paused for a retry topic. Using this class prevents breaking
+ * Spring Boot's auto configuration for other frameworks. Use this if you are using Spring
```

Comment:
This probably has to say that provided `TaskScheduler` must not be a bean.
Otherwise It is not clear why do we do the manual lifecycle control and how this would let us to avoid Spring Boot auto-configuration breakage.